execution_engine2.spec

```diff
@@ -97,47 +118,45 @@ module execution_engine2 {
         int wsid;
         string parent_job_id;
     } RunJobParams;
+
     /*
         Start a new job (long running method of service registered in ServiceRegistery).
         Such job runs Docker image for this service in script mode.
     */
     funcdef run_job(RunJobParams params) returns (job_id job_id) authentication required;
 
-
     /*
         Get job params necessary for job execution
     */
     funcdef get_job_params(job_id job_id) returns (RunJobParams params) authentication required;
+
     /*
-        is_started - optional flag marking job as started (and triggering exec_start_time
-            statistics to be stored).
-    */
-    /*
-    typedef structure {
-        job_id job_id;
-        boolean is_started;
-    } UpdateJobParams;
-    typedef structure {
-        list<string> messages;
-    } UpdateJobResults;
-    funcdef update_job(UpdateJobParams params) returns (UpdateJobResults)
-        authentication required;
+        job_id - a job id
+        status - the new status to set for the job.
     */
-
     typedef structure {
         job_id job_id;
         string status;
     } UpdateJobStatusParams;
 
     funcdef update_job_status(UpdateJobStatusParams params) returns (job_id job_id)
         authentication required;
+
+    /*
+        line - string - a string to set for the log line.
+        is_error - int - if 1, then this line should be treated as an error, default 0
+        ts - string - a timestamp for the log line (optional)
+
+        @optional ts
```

Comment:
Does the @optional keyword do anything?

Reply:
It's mostly a keyword for workspace objects, but it's generally in the KIDL language to mark a field as optional. Maybe one day in the far future when validation is automagically done, it'll be more necessary.